### PR TITLE
Use new right to work flow

### PR DESCRIPTION
**Pull request by JonF1**
## Context
Use changes made to the international candidate right to work flow hidden behind feature flag and make some small changes

## Changes proposed in this pull request
* Bypass the follow-up questions about immigration route if candidate selects no on right to work page and cut straight to review page. Keep those behind feature flag
* Replace right to work controller/form with immigration right to work behind feature flag and wire up from the routes down
* Update copy on various pages and components including visa status component

## Guidance to review
Visit personal details and inspect new flow. Select both yes/no and EU/non-EU/Brit/Irish nationalities. Behaviour should be the same as the prototype. Seed data on dev might throw up the 'English as a main language' question, but in prod candidates shouldn't see this unless they're carried over from last cycle and are from before the question was removed.
## Link to Trello card
https://trello.com/c/ZLhvqAH2/4387-move-immigration-status-from-a-conditionally-revealed-question-to-a-separate-question-page-half-done-under-a-feature-flag
https://trello.com/c/3UHHdgsv/4388-display-visa-sponsorship-warning-text-on-course-choices-if-applicable
https://trello.com/c/09m6QAfF/4386-update-content-on-right-to-work-or-study-in-the-uk-question-page

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by JonF1** [comment]
[Update content on 'Right to work or study in the UK' question page](https://trello.com/c/09m6QAfF/4386-update-content-on-right-to-work-or-study-in-the-uk-question-page)

**Comment by JonF1** [comment]
[Display visa sponsorship warning text on course choices if applicable](https://trello.com/c/3UHHdgsv/4388-display-visa-sponsorship-warning-text-on-course-choices-if-applicable)

File: app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+<h1 class="govuk-heading-l">
+  <%= t('page_titles.right_to_work') %>
+</h1>
+
+<p class="govuk-body">
+  You’ll need the right to work or study in the UK for the duration of your course.
+</p>
+
+<p class="govuk-body">
+  <%= govuk_link_to 'Check if you\'ve already got the right to work or study in the UK, or whether you’ll need to apply for a visa', t('train_to_teach_in_england.url') %>.
```
Comment (EmmaFrith) [inline app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb:12]: could we link to the relevant section of the guidance instead? here:

https://www.gov.uk/government/publications/train-to-teach-in-england-non-uk-applicants/train-to-teach-in-england-if-youre-a-non-uk-citizen#visa

Comment (EmmaFrith) [inline app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb:12]:
```suggestion
  <%= govuk_link_to 'Check whether you need to apply for a visa', t('train_to_teach_in_england.url') %>.
```

File: spec/presenters/candidate_interface/personal_details_review_presenter_spec.rb [hunk 2]
```diff
@@ -178,47 +177,42 @@ def rows(
...
         :nationalities_form,
         first_nationality: 'German',
       )
 
       application_form = build(
         :application_form,
-        recruitment_cycle_year: 2022,
         immigration_right_to_work: true,
         immigration_status: 'other',
         immigration_status_details: 'I have permanent residence',
```
Comment (EmmaFrith) [inline spec/presenters/candidate_interface/personal_details_review_presenter_spec.rb:218]:
```suggestion
        immigration_status_details: 'I have indefinite leave to remain in the UK',
```

Comment (EmmaFrith) [reply spec/presenters/candidate_interface/personal_details_review_presenter_spec.rb:218]: I think this is the official policy line to use

File: app/components/candidate_interface/course_choices_review_visa_status_component.html.erb [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
+      If you need a Student or Skilled Worker visa to get the right to work or study in the UK, you should find a
+      <%= govuk_link_to('find a course', t('find_postgraduate_teacher_training.production_url')) %> where sponsorship is available.
+    </p>
 
-    <ul class="govuk-list govuk-list--bullet">
-      <li>find a course that can sponsor student visas</li>
-      <li>find out about <%= govuk_link_to('other visas that allow you to study in the UK', t('train_to_teach_in_england.url')) %></li>
-    </ul>
+    <p class="govuk-body">
+      Sponsorship is not available for this course.
```
Comment (EmmaFrith) [inline app/components/candidate_interface/course_choices_review_visa_status_component.html.erb:17]:
```suggestion
```

Comment (EmmaFrith) [reply app/components/candidate_interface/course_choices_review_visa_status_component.html.erb:17]: just realised we say this in the heading of the warning text so don't need to repeat it

File: app/components/candidate_interface/course_choices_review_visa_status_component.html.erb [hunk 4]
```diff
@@ -1,18 +1,20 @@
 <% if can_sponsor_visa? %>
   <%= title %>
 <% else %>
-  <%= govuk_inset_text(classes: 'app-inset-text--narrow-border app-inset-text--important') do %>
+  <%= govuk_inset_text(classes: 'app-inset-text--narrow-border app-inset-text--important govuk-!-margin-top-0 govuk-!-padding-bottom-1') do %>
     <p class="govuk-heading-s app-inset-text__title"><%= title %></p>
 
     <p class="govuk-body">
-      You said that you do not yet have the right to work or study in the UK.
+      Check your eligibility for different <%= govuk_link_to('visa types and immigration statuses', t('train_to_teach_in_england.url')) %>.
```
Comment (EmmaFrith) [inline app/components/candidate_interface/course_choices_review_visa_status_component.html.erb:8]: Can we link to the place in the guidance which helps the candidate with this? e.g. here: https://www.gov.uk/government/publications/train-to-teach-in-england-non-uk-applicants/train-to-teach-in-england-if-youre-a-non-uk-citizen#visa

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Tested this in the review app. Works well!

Only spotted one minor bug, which is that when not selecting a radio option here, and pressing 'save and continue':

Screenshot 2022-02-14 at 13 22 58

...then the error screen selects the last option and suggests you enter some text. Instead, none of the radios should be selected, and the error message should ask you to select one with the wording "Select your immigration status" (or similar).

Screenshot 2022-02-14 at 13 25 46

File: app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb (hunk 1 above)
Comment (JonF1) [reply app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb]: thanks, fixed!

File: spec/presenters/candidate_interface/personal_details_review_presenter_spec.rb (hunk 2 above)
Comment (JonF1) [reply spec/presenters/candidate_interface/personal_details_review_presenter_spec.rb]: thanks

File: app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (JonF1) [reply app/views/candidate_interface/personal_details/immigration_right_to_work/_shared_form.html.erb]: thanks

**Comment by EmmaFrith** [comment]
Hm, for some reason, it still says 'I have permanent residence' here:

Screenshot 2022-02-16 at 17 08 01

@JonF1

**Comment by EmmaFrith** [comment]
@JonF1 could I be incredibly annoying and suggest we switch these 2 paras around? 

Screenshot 2022-02-16 at 17 20 21

**Comment by avinhurry** [comment]
**Closing this PR as it's been superseded by:**

- https://trello.com/c/uBlI9FrL/4501-update-content-on-the-right-to-work-or-study-page
- https://trello.com/c/VGbMj72b/4500-ensure-help-inset-text-still-appears
- https://trello.com/c/9qNLUUww/4525-add-new-what-is-your-immigration-status-page
- https://trello.com/c/oUWLYqAJ/4499-remove-question-from-right-to-work-flow
- https://trello.com/c/qLPBzUpR/4526-delete-immigration-entry-date-and-restructured-immigration-status-feature-flags

**Pull request closed without merging**